### PR TITLE
Refactor HttpResponseHandler to consolidate methods 

### PR DESCRIPTION
**Pull request by somln**
**Removed duplicate logic in the methods of the HttpResponseHandler.**
Since `Class<T>` implements `Type`, passing `Class<T>` to a method that accepts `Type` works seamlessly. Therefore, I consolidated the methods into a single one that accepts `Type` as a parameter, reducing duplicate code.

### CODE REVIEW

**Comment by somln** [comment]
@suyeon-jung-dev @jemlog  @seedspirit Can you please take a look?

File: src/main/java/io/litmuschaos/util/HttpResponseHandler.java [hunk 1]
```diff
@@ -26,21 +29,18 @@ public <T> T handleResponse(Response response, Class<T> responseType) throws Api
...
-            throw new ApiException(response);
-        }
-        if (response.body() == null) {
-            return transform("", responseType);
+    private <T> T transform(String responseBody, Type responseType) {
+        JsonElement jsonElement = JsonParser.parseString(responseBody);
+
+        if (jsonElement.isJsonObject()) {
+            JsonObject jsonObject = jsonElement.getAsJsonObject();
+            if (jsonObject.has("data")) {
```
Comment (suyeon-jung-dev) [inline src/main/java/io/litmuschaos/util/HttpResponseHandler.java:37]: Good work @somln ! 🚀

However, I believe that the process of checking the `data` field in the `jsonObject` and parsing it is a new feature. Could you separate this functionality into a different PR? That way, we can keep the scope of this one more focused.

**Review by suyeon-jung-dev — CHANGES_REQUESTED** [review]

File: src/main/java/io/litmuschaos/util/HttpResponseHandler.java (hunk 1 above)
Comment (somln) [reply src/main/java/io/litmuschaos/util/HttpResponseHandler.java:37]: Got it! I will modify it to remove that logic.

**Review by jemlog — APPROVED** [review]
LGTM🚀

**Comment by seedspirit** [comment]
LGTM, but it looks like you need to check DCO. I'll leave an article how to solve it.

https://github.com/litmuschaos/litmus-java-sdk/pull/16/checks

https://6991httam.medium.com/git-signed-off-commit-22838a21cf80

**Review by suyeon-jung-dev — APPROVED** [review]
Thanks to resolve it!🚀

**Review by namkyu1999 — APPROVED** [review]
